packages/@o3r/configuration/schematics/configuration-to-component/index.ts

```diff
@@ -306,9 +306,121 @@ export function ngAddConfigFn(options: NgAddConfigSchematicsSchema): Rule {
         }
       ]);
 
+      const updateComponentRulesWithSignal: Rule = chain([
+        addImportsRule(componentPath, [
+          {
+            from: '@angular/core',
+            importNames: [
+              'inject',
+              'input'
+            ]
+          },
+          {
+            from: '@o3r/configuration',
+            importNames: [
+              'configSignal',
+              'O3rConfig',
+              'ConfigurationBaseService',
+              'DynamicConfigurableWithSignal'
+            ]
+          },
+          {
+            from: `./${properties.name}.config`,
+            importNames: [
+              `${properties.configKey}_DEFAULT_CONFIG`,
+              `${properties.configKey}_CONFIG_ID`,
+              properties.componentConfig
+            ]
+          }
+        ]),
+        () => {
+          const componentSourceFile = ts.createSourceFile(
+            componentPath,
+            tree.readText(componentPath),
+            ts.ScriptTarget.ES2020,
+            true
+          );
+
+          const result = ts.transform(componentSourceFile, [
+            addInterfaceToClassTransformerFactory(`DynamicConfigurableWithSignal<${properties.componentConfig}>`, isO3rClassComponent),
+            (ctx) => (rootNode: ts.Node) => {
+              const { factory } = ctx;
+              const visit = (node: ts.Node): ts.Node => {
+                if (ts.isClassDeclaration(node) && isO3rClassComponent(node)) {
+                  const propertiesToAdd = generateClassElementsFromString(`
+  private readonly configurationService = inject(ConfigurationBaseService);
+
+  public config = input<Partial<${properties.componentConfig}>>();
+
+  @O3rConfig()
+  public readonly configSignal = configSignal(this.config, ${properties.configKey}_CONFIG_ID, ${properties.configKey}_DEFAULT_CONFIG, this.configurationService);`);
+
+                  const o3rDecorator = (ts.getDecorators(node) || []).find(isO3rClassDecorator)!;
```

Comment:
the `|| []` and the `!` don't seem to fit together

Reply:
I can do if you want `const o3rDecorator = ts.getDecorators(node)!.find(isO3rClassDecorator)!;`
Because I have check above to be sure that `node` is a class with the `O3rComponent` decorator (`isO3rClassComponent(node)`)